draft-ietf-dnsop-svcb-https.md

```diff
@@ -1133,24 +1137,18 @@ identifies the desired service.
 
 ## Client behavior {#ech-client-behavior}
 
-The general client behavior specified in {{client-behavior}} permits clients
-to retry connection with a less preferred alternative if the preferred option
-fails, including falling back to a direct connection if all SVCB options fail.
-This behavior is
+The SVCB-optional client behavior specified in {{client-behavior}} permits clients
+to fall back to a direct connection if all SVCB options fail.  This behavior is
 not suitable for ECH, because fallback would negate the privacy benefits of
-ECH.  Accordingly, ECH-capable clients SHALL implement the following
+ECH.  Accordingly, ECH-capable SVCB-optional clients MUST implement the following
 behavior for connection establishment:
 
-1. Perform connection establishment using HTTPS RRs as described in
-   {{client-behavior}}.  After step 4, if there were compatible HTTPS RRs,
-   they all had an "echconfig" key, and attempts to connect to them all failed,
-   terminate connection establishment.
-2. If the client implements Alt-Svc, try to connect using any entries from
-   the Alt-Svc cache.
-3. Continue connection establishment as in {{client-behavior}} if necessary.
+1. Perform SVCB resolution as described in {{client-behavior}}.
+2. If SVCB resolution succeeded, and all alternative endpoints have an "echconfig"
+   key, use SVCB-reliant connection establishment.
```

Comment:
Does this mean that ECH-capable SVCB-optional clients can't use Alt-Svc until we define an Alt-Svc-bis that addresses relevant issues?

Reply:
No.  ECH-capable SVCB-optional clients can use Alt-Svc unless all returned ServiceForm records include `echconfig`.

Reply:
Is there a reason we're missing what had been the previous steps 2+3?  (ie, to handle the case where the conditional in step 2 doesn't hold?)

Reply:
It's effectively moved to line 1054 (in the Alt-Svc section, instead of talking about Alt-Svc in the ECH section).

Reply:
In-general this numbered list is still confusing since it doesn't have an "else" clause to 2.  Perhaps switching it to be a bulletted list or just some flat text would be better?

Reply:
Perhaps:

Accordingly, ECH-capable SVCB-optional clients MUST use SVCB-reliant connection establishment if SVCB resolution succeeded (following {{client-behavior}}), and if all alternative endpoints have an "echconfig" key.

Reply:
OK, done.